include/zephyr/net/wifi_mgmt.h

```diff
@@ -1594,6 +1594,14 @@ struct wifi_mgmt_ops {
 	 * @return 0 if ok, < 0 if error
 	 */
 	int (*btm_query)(const struct device *dev, uint8_t reason);
+
+	/** Check if ap support Neighbor Report or not.
+	 * @param dev Pointer to the device structure for the driver instance.
+	 *
+	 * @return 1 if support, 0 is not support
+	 */
+	int (*bss_support_neighbor_rep)(const struct device *dev);
```

Comment:
As we return 0/1, I suggest we change the return value too bool to indicate that.